### PR TITLE
Video manager on frontend

### PR DESCRIPTION
**Pull request by rafaellehmkuhl**
- Show list of available video devices;
- Show list of available video streams per device;
- Allow creating new video stream for given device;
- Allow configuration of video device controls;
- Allow deleting available video streams;
- Allow viewing video stream details.

Includes UDP stream endpoint validation.

image

Completely solves #447

PS: I'm currently tweaking the interface yet. The thumbnail for example is a mess. Suggestions are much appreciated.

### CODE REVIEW

File: core/frontend/src/components/video-manager/VideoStreamCreationDialog.vue [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+      const supported_formats = this.device.formats.filter((format) => typeof format.encode === 'string')
+      return supported_formats.map((format) => ({
+        text: format.encode, value: format.encode,
+      }))
+    },
+    available_sizes(): {text: string, value: Size}[] {
+      if (this.selected_format === null) {
+        return []
+      }
+      return this.selected_format.sizes.map((size) => ({
```
Comment (patrickelectric) [inline core/frontend/src/components/video-manager/VideoStreamCreationDialog.vue:151]: We should also sort it by size to start with the highest resolution

File: core/frontend/src/components/video-manager/VideoStreamCreationDialog.vue [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+          <v-select
+            v-model="selected_interval"
+            :items="available_framerates"
+            :label="framerate_selector_label"
+            required
+          />
+
+          <v-text-field
+            v-model="stream_endpoint"
+            label="Stream endpoint"
```
Comment (patrickelectric) [inline core/frontend/src/components/video-manager/VideoStreamCreationDialog.vue:45]: This should have a default value for udp://0.0.0.0:5600

File: core/frontend/src/components/video-manager/StreamDetailsDialog.vue [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  <v-dialog
+    width="500"
+    :value="show"
+    @input="showDialog"
+  >
+    <v-card
+      width="100%"
+      class="pa-4 my-4"
+    >
+      <div class="d-flex flex-no-wrap align-center justify-space-between">
```
Comment (patrickelectric) [inline core/frontend/src/components/video-manager/StreamDetailsDialog.vue:11]: This card misses what is the encode type

File: core/frontend/src/components/video-manager/VideoControlsDialog.vue [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    width="500"
+    :value="show"
+    @input="showDialog"
+  >
+    <v-card>
+      <v-card-title>
+        Video configuration
+      </v-card-title>
+
+      <v-card-text class="d-flex flex-column">
```
Comment (patrickelectric) [inline core/frontend/src/components/video-manager/VideoControlsDialog.vue:12]: This also needs a sets to default method to set all values to the default

File: core/frontend/src/components/video-manager/VideoStream.vue [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        <v-card-title>{{ stream.video_and_stream.name }}</v-card-title>
+        <v-card-subtitle>{{ settings_summary }}</v-card-subtitle>
+      </div>
+      <div>
+        <v-btn
+          class="ma-2"
+          @click="openDetailsDialog"
+        >
+          <v-icon>mdi-eye</v-icon>
+          View details
```
Comment (patrickelectric) [inline core/frontend/src/components/video-manager/VideoStream.vue:17]: I think that this defails card is not necessary, we can have all the information on the main card:

**Review by patrickelectric — CHANGES_REQUESTED** [review]
Some UI chages are necessary and I'm unable to see the video locally, it also appears that there is a problem on the frontend when a h264 video is created.

**Comment by patrickelectric** [comment]
We should also add a more information button, like the wifi, to have the full information about the camera capabilities if possible.

**Review by patrickelectric — CHANGES_REQUESTED** [review]
Sizes are not sorted
image

The layout is weird
image


Details is not opening a popup when using fake source
Restaure defaults works for some but not for others, it's a bit weird, needs to click multiple times

**Comment by rafaellehmkuhl** [comment]
@patrickelectric they are sorted but reversed :smile:

**Comment by Williangalvani** [comment]
Can we get an "edit stream" option here?
image

**Comment by rafaellehmkuhl** [comment]
> We should also add a more information button, like the wifi, to have the full information about the camera capabilities if possible.
> 

It's a cool idea. I've opened #495 to track this future change.

**Review by patrickelectric — APPROVED** [review]

**Comment by patrickelectric** [comment]
> Can we get an "edit stream" option here?

#495

**Pull request merged**